doc/source/user_guide/local-usage.rst

```diff
@@ -0,0 +1,95 @@
+.. _local_usage:
+
+.. currentmodule:: nomenclature
+
+Local usage of a project
+========================
+
+.. attention:: This page is intended for users who are familiar with Python and
+    `git <https://git-scm.com>`_ (or a service like `GitHub <https://github.com>`_).
+
+You can use the **nomenclature** package locally (on your machine) for validation
+and region-aggregration. This can be helpful as part of processing your model results,
+or to ensure that a submission to an IIASA Scenario Explorer instance will succeed.
+
+Requirements
+------------
+
+1. Install the **nomenclature** package (see the :ref:`installation` instructions)
+2. Have a project folder that has the required :ref:`dir_structure`:
+
+   * Set up the folder with the required structure and files yourself, or
+   * Clone the git repository for a specific project
+     (e.g., `openENTRANCE <https://github.com/openENTRANCE/openentrance>`_)
+
+   .. attention:: When using a project repository from GitHub or a similar service,
+          make sure that you keep your local clone in sync with the upstream repository.
+
+Usage options
+-------------
+
+Validation against the codelists
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The easiest use case is to validate that a data file or an :class:`IamDataFrame <pyam.IamDataFrame>`
+is compatible with the codelists (lists of variables and regions) of a project's
+:class:`DataStructureDefinition`.
+
+If the :func:`process` function returns an :class:`IamDataFrame <pyam.IamDataFrame>`,
+the validation was successful. Otherwise, it will raise an error with details.
+
+.. code-block:: python
+
+  import pyam
+  from nomenclature import DataStructureDefinition, process
+
+  # Initialize the DataStructureDefinition from a suitable directory
+  dsd = DataStructureDefinition("definitions")
+
+  # Read the data using pyam
+  df = pyam.IamDataFrame("/path/to/file")
+
+  # Perform the validation
+  df = process(df, dsd)
```

Comment:
For demonstrating just validation maybe using `DataStructureDefinition.validate()` would be more straightforward. On the other hand using the `process` function throughout for **both** validation and the complete processing also has its advantages.